x-pack/plugin/ilm/src/main/java/org/elasticsearch/xpack/ilm/PolicyStepsRegistry.java

```diff
@@ -309,8 +309,11 @@ public Step getStep(final IndexMetadata indexMetadata, final Step.StepKey stepKe
         // n.b. we're using instance equality here for the IndexMetadata rather than object equality because it's fast,
         // this means that we're erring on the side of cache misses (if the IndexMetadata changed in any way, it'll be
         // a new instance, so we'll miss-and-repopulate the cache for the index in question)
-        if (cachedStep != null && cachedStep.v1() == indexMetadata && cachedStep.v2().getKey().equals(stepKey)) {
-            return cachedStep.v2();
+        if (cachedStep != null && cachedStep.v1() == indexMetadata) {
+            assert cachedStep.v2() != null;
```

Comment:
Can you add an assert error message to this? (it's so much nicer than getting back a generic `AssertionError`)

Reply:
Sure thing, 7ce8944653167ea3dee62ec6778ca6cbb664edc6